edit-post/hooks/block-list-change-detection/index.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import { differenceBy, camelCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { doAction } from '@wordpress/hooks';
+import { subscribe, select } from '@wordpress/data';
+
+/**
+ * A compare helper for lodash's difference by
+ *
+ * @param {Component} block A block object.
+ * @return {string} block.uid The uid of the block object for comparison.
+ */
+const compareBlocks = ( block ) => {
+	return block.uid;
+};
+
+/**
+ * A change listener for blocks
+ *
+ * The subscribe on the 'core/editor' getBlocks() function fires on any change,
+ * not just additions/removals. Therefore we actually compare the array with a
+ * previous state and look for changes in length or uid.
+ *
+ * @param  {function} selector Selector.
+ * @param  {function} listener Listener.
+ * @return {function}          Listener creator.
+ */
+const onBlocksChangeListener = ( selector, listener ) => {
+	let previousBlocks = selector();
+	return () => {
+		const selectedBlocks = selector();
+
+		// For performance reasons we first check the cheap length change,
+		// before we actuallly do a deep object comparison
+		if ( selectedBlocks.length !== previousBlocks.length ) {
+			// The block list length has changed, so there is obviously a change event happening
+			listener( selectedBlocks, previousBlocks );
+			previousBlocks = selectedBlocks;
+		} else if ( differenceBy( selectedBlocks, previousBlocks, compareBlocks ).length ) {
+			// A deep inspection has shown, that the list has changed
+			listener( selectedBlocks, previousBlocks, differenceBy( selectedBlocks, previousBlocks, compareBlocks ) );
+			previousBlocks = selectedBlocks;
+		}
+	};
+};
+
+/**
+ * Subscribe to block data
+ *
+ * This function subscribes to block data, compares old and new states upon
+ * change and fires actions accordingly.
+ */
+subscribe( onBlocksChangeListener( select( 'core/editor' ).getBlocks, ( blocks, oldBlocks, difference = null ) => {
```

Comment:
Should i add it to the documentation or change the variable name? Technically it is an intersection between blocks & oldBlocks, verbally it is the difference between these two arrays. (To signify that there is a difference, even though the length is equal.)

Reply:
What kind of value would it have? I don't understand what it means to be "the difference between these two arrays"... does that mean it's all the elements in either array not shared or just a boolean that signifies "yes these arrays are different"?

Docs around it would be great but if it's a boolean that is `true` when the arrays are different and `false` when not, something like `blocksHaveBeenChanged` or something might better? 🤷‍♂️

Reply:
Ah, I see the point. Currently `difference ` is an array of blocks that have been added. That signals that when the length of the blocklist stays equal, but there is a block in difference, that there was a transformation event. 

I have changed the var to a boolean value `hasChanged` to signal that.